AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/Statement+Model.swift

```diff
@@ -88,9 +89,39 @@ extension Statement: StatementModelConvertible {
                 from: value,
                 fieldType: field.type
             )
-            modelDictionary.updateValue(modelValue, forKeyPath: column)
+
+            // Check if the value for the primary key is `nil`. This is when an associated model does not exist.
+            // To create a decodable `modelDictionary` that can be decoded to the Model types, the entire
+            // object at this particular key should be set to `nil`. The following code does that by dropping the last
+            // path from the column, for example given "blog.id" `column` has a nil `modelValue`, then store the
+            // keypath in `skipColumns`, which will be used to set modelDictionary["blog"] to nil later on.
+            //
+            // `skipColumns` keeps track of these scenarios. At this point in the code we cannot make an assumption
+            // about the ordering of the columns, it may handle "blog.description", then "blog.id", then "blog.title".
+            // The code will perform the following:
+            //  1. set ["blog"]["description"] = `nil`, because it has not encountered `nil` primary key
+            //  2. store skipColumn["blog"], because the value is the primary key and is nil
+            //  3. skip setting modelDictionary["blog"]["title"], because `skipColumn` has been set for "blog".
+            if field.isPrimaryKey && modelValue == nil {
```

Comment:
- primary key usage here, rebase with CPK work to make sure tests are still passing